src/orders/components/OrderHistory/utils.tsx

```diff
@@ -37,6 +56,73 @@ const timelineEventTypes = {
 };
 
 export const isTimelineEventOfType = (
-  type: "extendable" | "secondaryTitle" | "rawMessage" | "note" | "linked",
+  type:
+    | "extendable"
+    | "secondaryTitle"
+    | "rawMessage"
+    | "note"
+    | "linked"
+    | "discount",
   eventType: OrderEventsEnum
 ) => !!timelineEventTypes[type]?.includes(eventType);
+
+export const isTimelineEventOfDiscountType = (eventType: OrderEventsEnum) =>
+  isTimelineEventOfType("discount", eventType);
+
+const selectEmployeeName = ({
+  firstName,
+  lastName,
+  email
+}: OrderDetails_order_events_user) => {
+  const hasFirstNameAndSurname = !!firstName && !!lastName;
+  const hasOnlyFirstName = !!firstName && !lastName;
+
+  if (hasFirstNameAndSurname) {
+    return `${firstName} ${lastName}`;
+  }
+
+  if (hasOnlyFirstName) {
+    return firstName;
+  }
```

Comment:
Since both firstName and lastName are always of type `String`:

```suggestion
  if (firstName) {
    return `${firstName} ${lastName}`.trim();
  }
```